Nesc/Controller/Controller.php

```diff
@@ -0,0 +1,16 @@
+<?php
+namespace Controller;
+require_once __DIR__.'/Traits/Helpers.php';
+use Helpers;
+
+class Controller {
+    use Helpers;
+
+    public  function run($value){
+        $methodName = $this->methodName($value);
+        $controllerFile = $this->ControllerFile($value);
```

Comment:
i believe  run method should get controller name and method name as parameters 
it shouldn't know how to parse the route string it's routing responsibility